### PR TITLE
fix booting from cdrom

### PR DESCRIPTION
**Pull request by lnussel**

### CODE REVIEW

**Comment by sysrich** [comment]
Thanks for this Ludwig

File: backend/qemu.pm [hunk 1]
```diff
@@ -418,29 +420,33 @@ sub start_qemu() {
...
-        for my $i (1 .. 6) {    # check for up to 6 ADDON ISOs
-            if ($vars->{"ISO_$i"}) {
-                my $addoniso = $vars->{"ISO_$i"};
-                push(@params, "-drive", "if=scsi,id=addon_$i,file=$addoniso,media=cdrom");
-            }
+        for my $k (sort grep { /^ISO_\d+$/ } keys %$vars) {
+            my $addoniso = $vars->{$k};
+            my $i = $k;
+            $i =~ s/^ISO_//;
+            push(@params, '-drive',  "media=cdrom,if=none,id=cd$i,format=raw,file=$addoniso");
```
Comment (dzedro) [inline backend/qemu.pm]: what is if=none, why not e.g. if=virtio ?

Comment (lnussel) [inline backend/qemu.pm]: https://github.com/qemu/qemu/blob/master/docs/qdev-device-use.txt

Comment (dzedro) [inline backend/qemu.pm]: thank you

File: backend/qemu.pm [hunk 2]
```diff
@@ -418,29 +420,33 @@ sub start_qemu() {
...
+
         if ($iso) {
             if ($vars->{USBBOOT}) {
                 push(@params, "-drive",  "if=none,id=usbstick,file=$iso,snapshot=on");
                 push(@params, "-device", "usb-ehci,id=ehci");
                 push(@params, "-device", "usb-storage,bus=ehci.0,drive=usbstick,id=devusb");
             }
-            elsif ($vars->{CDMODEL}) {
+            else {
                 push(@params, '-drive',  "media=cdrom,if=none,id=cd0,format=raw,file=$iso");
```
Comment (dzedro) [inline backend/qemu.pm]: I think snapshot=on is missing for UEFI

Comment (lnussel) [inline backend/qemu.pm]: indeed. meanwhile I found that adding snapshot=on here breaks loading cd drives again which is needed for zdup on uefi o_O.

**Comment by lnussel** [comment]
ok, I gave up on zdup with UEFI. we can only test that with NET iso then

**Pull request merged**